packages/compiler-cli/src/ngtsc/typecheck/src/type_check_block.ts

```diff
@@ -430,7 +431,8 @@ function tcbProcessTemplateDeclaration(tmpl: TmplAstTemplate, tcb: Context, scop
       // any template guards, and generate them if needed.
       dir.ngTemplateGuards.forEach(inputName => {
         // For each template guard function on the directive, look for a binding to that input.
-        const boundInput = tmpl.inputs.find(i => i.name === inputName);
+        const boundInput = tmpl.inputs.find(i => i.name === inputName) ||
+            tmpl.templateAttrs.filter(isBoundAttribute).find(i => i.name === inputName);
```

Comment:
```suggestion
            tmpl.templateAttrs.find((i: TextAttribute|BoundAttribute): i is BoundAttribute => i instanceof BoundAttribute && i.name === inputName);
```

This avoids the cost of `filter()` for every lookup while still narrowing the type correctly.